okio/jvm/src/main/java/okio/Buffer.kt

```diff
@@ -252,8 +252,9 @@ class Buffer : BufferedSource, BufferedSink, Cloneable, ByteChannel {
     return result
   }
 
+  @Throws(EOFException::class)
   override fun readByte(): Byte {
-    check(size != 0L) { "size == 0" }
+    if (size == 0L) throw EOFException()
```

Comment:
Would it make sense to keep the messages?

Reply:
I think IllegalStateException needs the message ’cause it could be anything. For EOFException the class name says it all.